File: TUF/Array/FindAllNonRptElm.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.Map.Entry;

public class FindAllNonRptElm {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        findNonRepeatingElements(arr);
    }

    private static void findNonRepeatingElements(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            map.put(arr[i], map.getOrDefault(arr[i], 0) + 1);
        }
        for (Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == 1) {
                System.out.print(entry.getKey() + ", ");
            }
        }
    }
}

File: TUF/Array/FindAllNonRptElm.ts
function findNonRepeatingElements(arr: number[]): void {
    let map = new Map<number, number>();
    for (let i: number = 0; i < arr.length; i++) {
        map.set(arr[i], (map.get(arr[i]) || 0) + 1);
    }
    let result: number[] = [];
    for (let [key, value] of map.entries()) {
        if (value == 1) {
            result.push(key);
        }
    }
    console.log(`All non repeating elements are: ${result}`);
}

let arr: number[] = [1, 2, 2, 3, 4, 3, 3, 4, 5, 6, 7];
findNonRepeatingElements(arr);

export { };